django/contrib/admin/static/admin/js/SelectBox.js

```diff
@@ -8,23 +8,56 @@
             const cache = SelectBox.cache[id];
             const boxOptions = box.options;
             const boxOptionsLength = boxOptions.length;
-            for (let i = 0, j = boxOptionsLength; i < j; i++) {
-                const node = boxOptions[i];
-                cache.push({value: node.value, text: node.text, displayed: 1});
+            const noGroup = [];
+            if(box.children.length > 0) {
+                // optgroup not present
+                if(box.children[0].tagName.toLowerCase() === 'option') {
+                    for (let i = 0, j = boxOptionsLength; i < j; i++) {
+                        const node = boxOptions[i];
+                        noGroup.push({value: node.value, text: node.text, displayed: 1 });
+                    }
+                    cache.push({ 'group': null, 'items': noGroup });
+                }
+                // optgroup present
+                if(box.children[0].tagName.toLowerCase() === 'optgroup') {
+                    for (let p = 0; p < box.children.length; p++) { 
+                        const options = box.children[p].children;
+                        const group = [];
+                        for (let r = 0; r < options.length; r++) {
```

Comment:
Can use `for ... of` here and elsewhere. See #12866.

Reply:
- replace `for` with `for ... of` except one, 
- I didn't find a way to replace with `for ---of` in delete_from_cache function, please suggest if possible